### ISSUE TITLE
[bug] `no-leaked-interval` rule gives a false positive when `setTimeout` timer is assigned to an uninitialized let

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#753]
### Describe the bug

The `no-leaked-interval` rule produces a false positive when the timer from `setTimeout` is assigned to a variable declared with let that has not been initialized
The same thing happens with `no-leaked-interval`

### Reproduction

```tsx
  import { useEffect, useRef } from "react";

  function Example() {
    useEffect(() => {
      let timeoutId1: number | null;
      let timeoutId2: number | null;
      if (Math.random() > 0.5) {
        timeoutId1 = setTimeout(() => {}, 1000);
        timeoutId2 = setTimeout(() => {}, 1000);
      }
      return () => {
        timeoutId1 && clearTimeout(timeoutId1);
        timeoutId2 && clearTimeout(timeoutId2);
      };
    }, []);
  }
```

### Expected behavior

No errors reported.

### Platform and versions

```Text
Node.js: v20.17.0
@eslint-react/react-plugin: 1.12.2
```
